src/components/structures/auth/ForgotPassword.js

```diff
@@ -89,6 +120,8 @@ module.exports = React.createClass({
     onSubmitForm: function(ev) {
         ev.preventDefault();
 
+        if (!this.state.serverIsAlive) return;
```

Comment:
Hmm, sorry - think we probably need to either not block form submission or have some way to re-check in all 3 places.